File: app/src/main/java/com/allen/appmanager/AppManagerAdapter.java
/*
 *
 *  * Copyright  2017 [AllenCoderr]
 *  *
 *  * Licensed under the Apache License, Version 2.0 (the "License");
 *  * you may not use this file except in compliance with the License.
 *  * You may obtain a copy of the License at
 *  *
 *  *     http://www.apache.org/licenses/LICENSE-2.0
 *  *
 *  * Unless required by applicable law or agreed to in writing, software
 *  * distributed under the License is distributed on an "AS IS" BASIS,
 *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  * See the License for the specific language governing permissions and
 *  * limitations under the License.
 *
 */

package com.allen.appmanager;

import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;

import java.util.List;

/**
 * 文 件 名: AppManagerAdapter
 * 创 建 人: Allen
 * 创建日期: 2017/4/17 16:16
 * 修改时间：
 * 修改备注：
 */

public class AppManagerAdapter extends BaseQuickAdapter<AppInfo, AppManagerAdapter.ViewHolder> {
    public AppManagerAdapter(int layoutResId, List<AppInfo> data) {
        super(R.layout.app_item, data);
    }

    public AppManagerAdapter(List<AppInfo> data) {
        super(R.layout.app_item, data);
    }


    @Override
    protected void convert(ViewHolder helper, AppInfo item) {
        helper.addOnClickListener(R.id.btCopy);
        helper.imgApp.setImageDrawable(item.getAppIcon());
        helper.apkName.setText(item.getAppLabel());
        helper.pkgName.setText(item.getPkgName());
        helper.apkSize.setText(item.getTotalSize());
        helper.apkVersion.setText(item.getmVersion());
        helper.sigmd5.setText(item.getSigmd5());
    }

    public static class ViewHolder  extends BaseViewHolder{
        public View rootView;
        public ImageView imgApp;
        public Button btCopy;
        public TextView apkName;
        public TextView pkgName;
        public TextView apkSize;
        public TextView apkVersion;
        public TextView sigmd5;

        public ViewHolder(View rootView) {
            super(rootView);
            this.rootView = rootView;
            this.imgApp = (ImageView) rootView.findViewById(R.id.imgApp);
            this.btCopy = (Button) rootView.findViewById(R.id.btCopy);
            this.apkName = (TextView) rootView.findViewById(R.id.apkName);
            this.pkgName = (TextView) rootView.findViewById(R.id.pkgName);
            this.apkSize = (TextView) rootView.findViewById(R.id.apkSize);
            this.apkVersion = (TextView) rootView.findViewById(R.id.apkVersion);
            this.sigmd5 = (TextView) rootView.findViewById(R.id.sigmd5);
        }

    }
}

File: kotlinapp/src/main/java/com/allen/kotlinapp/AppManagerAdapter.kt
package com.allen.kotlinapp

import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder

/**
 * 文 件 名: AppManagerAdapter
 * 创 建 人: Allen
 * 创建日期: 2017/6/12 16:57
 * 修改时间：
 * 修改备注：
 */

class AppManagerAdapter : BaseQuickAdapter<AppInfo, AppManagerAdapter.ViewHolder> {
    constructor(layoutResId: Int, data: List<AppInfo>) : super(R.layout.app_item, data) {}

    constructor(data: List<AppInfo>) : super(R.layout.app_item, data) {}


    override fun convert(helper: ViewHolder, item: AppInfo) {
        helper.addOnClickListener(R.id.btCopy)
        helper.imgApp.setImageDrawable(item.appIcon)
        helper.apkName.setText(item.appLabel)
        helper.pkgName.setText(item.pkgName)
        helper.apkSize.setText(item.totalSize)
        helper.apkVersion.setText(item.mVersion)
        helper.sigmd5.setText(item.sigmd5)
    }

    class ViewHolder(var rootView: View) : BaseViewHolder(rootView) {
        var imgApp: ImageView
        var btCopy: Button
        var apkName: TextView
        var pkgName: TextView
        var apkSize: TextView
        var apkVersion: TextView
        var sigmd5: TextView

        init {
            this.imgApp = rootView.findViewById(R.id.imgApp) as ImageView
            this.btCopy = rootView.findViewById(R.id.btCopy) as Button
            this.apkName = rootView.findViewById(R.id.apkName) as TextView
            this.pkgName = rootView.findViewById(R.id.pkgName) as TextView
            this.apkSize = rootView.findViewById(R.id.apkSize) as TextView
            this.apkVersion = rootView.findViewById(R.id.apkVersion) as TextView
            this.sigmd5 = rootView.findViewById(R.id.sigmd5) as TextView
        }

    }
}
